### PR TITLE
refactor(journey-maps): update MapLibre to V2

### PR DESCRIPTION
**Pull request by sebisteiner**

### CODE REVIEW

File: src/journey-maps/angular/types.d.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// TOOD ses: Why doesn't it work with https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/mapbox__point-geometry
+
+declare module '@mapbox/point-geometry' {
```
Comment (sebisteiner) [inline src/journey-maps/angular/types.d.ts:3]: @kyubisation I couldn't get it to work with the definitely typed types. (_Could not find a declaration file for module '@mapbox/point-geometry'_)

Do you have any idea or should I keep it like this for the moment?

**Comment by sebisteiner** [comment]
@kyubisation Can you help me with the failing build? I don't understand the problem yet...

File: src/journey-maps/angular/components/level-switch/services/map-layer-filter.ts [hunk 2]
```diff
@@ -31,20 +31,23 @@ export class SbbMapLayerFilter {
     this._setLayerVisibility('rokas_background_mask', level < 0);
   }
 
-  private _calculateLayerFilter(oldFilter: any[], level: number): any[] {
+  private _calculateLayerFilter(
+    oldFilter: FilterSpecification | void,
```
Comment (kyubisation) [inline src/journey-maps/angular/components/level-switch/services/map-layer-filter.ts:35]: void should only be used as a return value
Use undefined, if the value might be undefined

**Review by kyubisation — CHANGES_REQUESTED** [review]
One detail, other than that LGTM
After researching a bit more, I would recommend adding the peer dependencies and just have warnings for the web components consumers. What do you think?

**Comment by sebisteiner** [comment]
> One detail, other than that LGTM After researching a bit more, I would recommend adding the peer dependencies and just have warnings for the web components consumers. What do you think?

Ok for me.

File: src/journey-maps/package.json [hunk 3]
```diff
@@ -13,6 +13,12 @@
     "url": "https://github.com/sbb-design-systems/sbb-angular/issues"
   },
   "homepage": "https://angular.app.sbb.ch",
+  "peerDependencies": {
+    "@angular/common": "0.0.0-NG",
+    "@angular/core": "0.0.0-NG",
+    "@types/geojson": "7946",
+    "maplibre-gl": "2.1"
```
Comment (kyubisation) [inline src/journey-maps/package.json:20]: Should we maybe use a range? Or would you prefer more precise versions?
```suggestion
    "@types/geojson": ">=7946",
    "maplibre-gl": "^2.1.0"
```

**Review by kyubisation — APPROVED** [review]
LGTM 👍

File: src/journey-maps/package.json (hunk 3 above)
Comment (sebisteiner) [reply src/journey-maps/package.json]: 👍

**Pull request merged**